src/fraction/unicode_str_io.rs

```diff
@@ -0,0 +1,592 @@
+                            '1' => '¹',
+                            '2' => '²',
+                            '3' => '³',
+                            '4' => '⁴',
+                            '5' => '⁵',
+                            '6' => '⁶',
+                            '7' => '⁷',
+                            '8' => '⁸',
+                            '9' => '⁹',
+                            '0' => '⁰',
+                            _ => '?',
+                        })
+                        .collect::<String>(),
+                    r.denom()
+                        .to_string()
+                        .chars()
+                        .map(|c| match c {
+                            '1' => '₁',
+                            '2' => '₂',
+                            '3' => '₃',
+                            '4' => '₄',
+                            '5' => '₅',
+                            '6' => '₆',
+                            '7' => '₇',
+                            '8' => '₈',
+                            '9' => '₉',
+                            '0' => '₀',
+                            c => c,
+                        })
+                        .collect::<String>(),
+                )
+            }
+            _ => write!(f, "{}", self.0),
+        }
+    }
+}
+
+impl<T: Clone + Integer + fmt::Display> GenericFraction<T> {
+    /// Display the fraction using FRACTION SLASH '\u{2044}' '⁄'
+    /// If you have font support, this is the way Unicode wants to display fractions
+    /// ```
+    /// use fraction::Fraction;
+    /// assert_eq!(
+    ///   format!("{}", Fraction::new(1u8,2u8).get_unicode_display()),
+    ///   "1⁄2"
+    /// );
+    /// assert_eq!(
+    ///   format!("{}", Fraction::new(3u8,2u8).get_unicode_display()),
+    ///   "3⁄2"
+    /// );
+    /// ```
+    pub fn get_unicode_display(&self) -> UnicodeDisplay<'_, T> {
+        UnicodeDisplay(self)
+    }
+}
+
+impl<T: Clone + Integer + fmt::Display + From<u8>> GenericFraction<T> {
+    /// Parse a unicode string
+    /// The string can be:
+    /// - A normal fraction e.g. "1/2"
+    /// - A vulgar fraction e.g. '½'
+    /// - ~A mixed vulgar fraction "1½"~
+    /// - A unicode fraction e.g. "1⁄2" where '⁄' can be any of:
+    ///   - '/' ASCII SOLIDUS
+    ///   - '⁄' FRACTION SLASH
+    ///   - '∕' DIVISION SLASH
+    ///   - '÷' DIVISION SIGN
+    /// - A mixed unicode fraction: "1\u{2063}1⁄2": 1⁤1⁄2
+    ///   - '\u{2064}' INVISIBLE PLUS
+    ///   - '\u{2063}' INVISIBLE SEPARATOR
+    ///   - NOT ~'\u{2062}' INVISIBLE TIMES~
+    /// - A super-subscript fraction "¹/₂"
+    /// - A mixed super-subscript fraction  "1¹/₂"
+    ///
+    /// Focus is on being lenient towards input rather than being fast.
+    /// ```
+    /// use fraction::Fraction;
+    /// let v = vec![
+    ///  ("1/2", Fraction::new(1u8,2u8)),
+    ///  ("-1/2", Fraction::new_neg(1u8,2u8)),
+    ///  ("½", Fraction::new(1u8,2u8)),
+    ///  // ("1½", Fraction::new(1u8,2u8)),
```

Comment:
redundant commented values?

Reply:
Not entirely, I put them there, since it's a mixed vulgar fraction, which currently cannot be parsed. I'll add a comment to clarify